packages/api/src/users/users.repository.ts

```diff
@@ -1,26 +1,27 @@
-import { Injectable } from '@nestjs/common';
+import { Inject, Injectable } from '@nestjs/common';
+import { nanoid } from 'nanoid';
 
-import type { User } from '@coderscamp/shared/models/user';
+import type { NotRegisteredUser, RegisteredUser, User, UserId } from '@coderscamp/shared/models';
 
-import { PrismaService } from '../prisma/prisma.service';
+import { USER_REPOSITORY_PORT, UserRepositoryPort } from '../contracts/user.repository';
 
 @Injectable()
 export class UsersRepository {
```

Comment:
This class is not required, you can use UserRepositoryPort directly - it's an abstraction over your database access.